File: CPP/Program/Parkinglot.cpp
#pragma once
// Memasukan library yang digunakan
#include <iostream>
#include <string>
#include "Vehicle.cpp"

// Deklarasi namespace
using namespace std;

// Deklarasi kelas
class Parkinglot
{
private:
    // Atribut Privat
    string kapasitas;
    string jumlahSaatIni;
    // Composite dengan Vehicle
    Vehicle kendaraan;
public:
    // Constructor
    Parkinglot()
    {

    }

    //Constructor dengan Parameter
    Parkinglot(string kapasitas, string jumlahSaatIni, Vehicle kendaraan)
    {
        this->kapasitas = kapasitas;
        this->jumlahSaatIni = jumlahSaatIni;
        this->kendaraan = kendaraan;
    }

    // Setter dan Getter
    void setKapasitas(string kapasitas)
    {
         this->kapasitas = kapasitas;
    }

    void setJumlahSaatIni(string jumlahSaatIni)
    {
        this->jumlahSaatIni = jumlahSaatIni;
    }

    void setKendaraan(Vehicle kendaraan)
    {
        this->kendaraan = kendaraan;   
    }

    string getKapasitas()
    {
        return this->kapasitas;
    }

    string getjumlahSaatIni()
    {
        return this->jumlahSaatIni;
    }
    
    Vehicle getKendaraan()
    {
        return this->kendaraan;
    }

    // Destructor
    ~Parkinglot()
    {

    }
};

// Parkinglot::Parkinglot(/* args */)
// {
// }

// Parkinglot::~Parkinglot()
// {
// }

File: Python/Program/ParkingLot.py
## Saya Muhammad Rifky Afandi dengan NIM 2202346 mengerjakan LP2 dalam mata kuliah 
## Desain Pemorograman Berbasis Objek untuk keberkahanNya maka saya tidak 
## melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.

## Membuat class
class Parkinglot:
    ## Atribut privat
    kapasitas=""
    jumlahSaatIni=""
    kendaraan=None ## Composite
    
    ## Constructor
    ## Atribut Publik
    def __init__(self, kapasitas="", jumlahSaatIni="", kendaraan=None):
        self.kapasitas = kapasitas
        self.jumlahSaatIni = jumlahSaatIni
        self.kendaraan = kendaraan

    ## Getter dan Setter
    def setKapasitas(self, kapasitas):
        self.kapasitas = kapasitas

    def setJumlahSaatIni(self, jumlahSaatIni):
        self.jumlahSaatIni = jumlahSaatIni

    def setKendaraan(self, kendaraan):
        self.kendaraan = kendaraan

    def getKapasitas(self):
        return self.kapasitas

    def getJumlahSaatIni(self):
        return self.jumlahSaatIni

    def getKendaraan(self):
        return self.kendaraan